src/main/java/org/opensearch/knn/index/codec/transfer/OffHeapVectorTransfer.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.transfer;
+
+import lombok.Getter;
+import org.opensearch.knn.index.vectorvalues.KNNVectorValues;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * <p>
+ * The class is intended to transfer {@link KNNVectorValues} to off heap memory.
+ * It also provides and ability to quantize the vector before it is transferred to offHeap memory.
+ * The ability to quantize is added as to not iterate KNN {@link KNNVectorValues} multiple times.
+ * </p>
+ *
+ * <p>
+ * The class is not thread safe.
+ * </p>
+ *
+ * @param <T>  byte[] or float[]
+ */
+public abstract class OffHeapVectorTransfer<T> implements Closeable {
+
+    @Getter
+    private long vectorAddress;
+    protected final int transferLimit;
+
+    private final List<T> vectorsToTransfer;
+
+    public OffHeapVectorTransfer(final int transferLimit) {
+        this.transferLimit = transferLimit;
+        this.vectorsToTransfer = new ArrayList<>(transferLimit);
+        this.vectorAddress = 0;
+    }
+
+    public boolean transfer(T vector, boolean append) throws IOException {
```

Comment:
Please add java docs on all the public functions atleast on the base class functions so that it is easy to know what these functions are supposed to do.